src/server/pfs/fuse/fuse.go

```diff
@@ -13,12 +13,48 @@ import (
 
 	"github.com/pachyderm/pachyderm/v2/src/client"
 	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errutil"
 	"github.com/pachyderm/pachyderm/v2/src/internal/progress"
 	"github.com/pachyderm/pachyderm/v2/src/internal/uuid"
 )
 
 // Mount pfs to target, opts may be left nil.
 func Mount(c *client.APIClient, target string, opts *Options) (retErr error) {
+	if opts.RepoOptions == nil {
+		opts.RepoOptions = make(map[string]*RepoOptions)
+	}
+	if len(opts.RepoOptions) == 0 {
+		// Behavior of `pachctl mount` with no args is to mount everything. Make
+		// that explicit here before we pass in the configuration to avoid
+		// needing to special-case this deep within the FUSE implementation.
+		// (`pachctl mount-server` does _not_ have the same behavior. It mounts
+		// nothing to begin with.)
+		ris, err := c.ListRepo()
+		if err != nil {
+			return err
+		}
+		for _, ri := range ris {
+			// Behavior here is that we explicitly mount repos to mounts named
+			// by the repo name. This is different to `pachctl mount-server`
+			// which supports mounting different versions of the same repo at
+			// different named paths.
+			branch := "master" // XXX is the old implementation always defaulting to master?
+			bi, err := c.InspectBranch(ri.Repo.Name, branch)
+			if err != nil && !errutil.IsNotFoundError(err) {
+				return err
+			}
+			isOutputBranch := bi != nil && len(bi.Provenance) > 0
+			write := opts.Write
+			if isOutputBranch {
+				write = false
```

Comment:
could the above be just `write := opts.Write && (bi == nil || len(bi.Provenance) == 0)` (or `write := opts.Write && !isOutputBranch)` if you prefer the temporary variable)?

Reply:
yeah, i just thought this was easier to read